app/frontend/courses/curricula/components/CoursesCurriculum__Overlay.res

```diff
@@ -210,7 +228,18 @@ let loadTargetDetails = (target, currentUser, send, ()) => {
         reloadSubmissions(send, Target.id(target))
       }
 
-      resolve(targetDetails)
+      let hasCommentParam = DomUtils.hasUrlParam(~key="comment_id")
+      let hasSubmissionParam = DomUtils.hasUrlParam(~key="submission_id")
+
+      if hasCommentParam && hasSubmissionParam {
+        handleUrlParam(~key="comment_id", ~prefix="comment-", send, targetDetails)
+      }
+
+      if hasSubmissionParam && !hasCommentParam {
+        handleUrlParam(~key="submission_id", ~prefix="submission-", send, targetDetails)
+      }
+
+      resolve()
```

Comment:
This used to `resolve(targetDetails)`. I'm unsure why that was removed. 🤔

Reply:
It was a mistake